### DISCUSSION TITLE
Add `protected` to `no-exposed-subject` eslint rule

### DISCUSSION THREAD

**Discussion by butterknight** [infinum/js-linters#89] [General]
Hey!

The `no-exposed-subjects` rule by default (when turned on) requires `Subject`s to always be _private_. Its  [documentation](https://github.com/cartant/eslint-plugin-rxjs/blob/main/docs/rules/no-exposed-subjects.md) states that it can be configured to allow them to be _protected_ as well. The use-case for this are abstract classes that could expose a subject to its inheriting classes.

The downside of this is that, with the new version of Angular, _protected_ members can be accessed from the template so it could be a potential risk.

Do you think it might be beneficial enough to allow _protected_ with this rule or should we leave it only for _private_ as it is currently?

**Comment by thabalija** [comment]
We use protected when we have class inheritance. I think that the proper way would be to have:
- `private` `Subject` in the base class
- a `protected` method in a base class that updated that subject
- public `Observable` which is used in the template (either in base or class which extends it, depending on the needs)

This approach has a bit of a boilerplate, but I think it is "by the book" solution.

**Reply by fvoska** [reply]
I think we could allow protected subjects. This will help with inheritance and in Angular v14+ it will also work if you need just a simple subject for use in template. Since Angular 14, you can access protected members from template: https://twitter.com/angular/status/1537181658461265920?lang=en

**Reply by butterknight** [reply]
By default I'd be fully up for it, but I'm just a bit concerned whether allowing _protected_ could be a potential risk precisely because they can now be used in the template.

**Reply by butterknight** [reply]
Is this something on what we can make a decision or do we need to discuss it further?

**Comment by fvoska** [comment]
I vote for leaving it private-only

**Reply by butterknight** [reply]
I tend to agree. Should we close this discussion?

How do we actually close [this discussion](https://github.com/infinum/js-linters/discussions/89)? 😅

**Reply by fvoska** [reply]
![image](https://user-images.githubusercontent.com/3170728/205074465-59e1f2a3-0d06-4441-829d-446c39fb332a.png)

**Reply by fvoska** [reply]
I have locked the discussion
